packages/core-ethereum/crates/core-ethereum-misc/src/transaction_queue.rs

```diff
@@ -0,0 +1,147 @@
+use crate::errors::CoreEthereumError::TransactionSubmissionFailed;
+use crate::errors::Result;
+use async_lock::RwLock;
+use async_std::channel::{bounded, Receiver, Sender};
+use async_trait::async_trait;
+use core_ethereum_db::traits::HoprCoreEthereumDbActions;
+use core_types::acknowledgement::AcknowledgedTicket;
+use std::sync::Arc;
+use utils_log::{error, warn};
+
+/// Enumerates all possible outgoing transactions
+#[derive(Clone, PartialEq, Debug)]
+pub enum Transaction {
+    RedeemTicket(AcknowledgedTicket),
+}
+
+/// Implements execution of each `Transaction` and also **awaits** its confirmation.
+#[cfg_attr(test, mockall::automock)]
+#[async_trait(? Send)]
+pub trait TransactionExecutor {
+    async fn redeem_ticket(&self, ticket: AcknowledgedTicket) -> Result<()>;
+}
+
+/// Represents a result of an Ethereum transaction after it has been confirmed.
+#[derive(Clone, Copy, Debug)]
+pub struct TransactionResult;
+
+/// Notifies about completion of a transaction (success or failure).
+pub type TransactionCompleted = async_oneshot::Receiver<TransactionResult>;
```

Comment:
Would it not be better to use the native async oneshot channel? https://docs.rs/futures/latest/futures/channel/oneshot/index.html